### PR TITLE
Redirect incoming URLs with short languages to the long-language variant

### PR DESCRIPTION
**Pull request by diox**
Pre-requisite for https://github.com/mozilla/addons/issues/15232

### Context

That was previously only done for Accept-Language HTTP header in addons-server, not in URLs, despite being supported on addons-frontend.

### Testing

See unit test, you can try those URLs/redirects locally too.

### CODE REVIEW

File: src/olympia/amo/urlresolvers.py [hunk 1]
```diff
@@ -36,8 +36,15 @@ def split_path(path_):
...
-        if first_lower in settings.LANGUAGE_URL_MAP:
+        # First test shorter languages shortcuts.
+        if not dash and first in settings.SHORTER_LANGUAGES:
+            first = settings.SHORTER_LANGUAGES[first]
+            if second in amo.APPS:
+                return first, second, rest
+            else:
+                return first, '', first_rest
+        # Then check language-territory.
+        elif first_lower in settings.LANGUAGE_URL_MAP:
```
Comment (KevinMind) [inline src/olympia/amo/urlresolvers.py:47]: Nit: the elif could be returned to an if since the first if definitely returns right? I tend to prefer split if statements if they all return since elif makes me think we could continue executing after the final branch, but in this case there is a (not totally obvious) return

Comment (KevinMind) [reply src/olympia/amo/urlresolvers.py:47]: but I guess there are already other if/elif/elif branches so nvm 🤷

File: src/olympia/amo/urlresolvers.py [hunk 2]
```diff
@@ -36,8 +36,15 @@ def split_path(path_):
         first_lower = first.lower()
         lang, dash, territory = first_lower.partition('-')
 
-        # Check language-territory first.
-        if first_lower in settings.LANGUAGE_URL_MAP:
+        # First test shorter languages shortcuts.
+        if not dash and first in settings.SHORTER_LANGUAGES:
+            first = settings.SHORTER_LANGUAGES[first]
```
Comment (KevinMind) [inline src/olympia/amo/urlresolvers.py:41]: Nit: the "first" renaming is blowing up my brain. I know what it's doing but it feels confusing. could you use a different name "first_short" for example?

File: src/olympia/amo/tests/test_url_prefix.py [hunk 3]
```diff
@@ -38,6 +38,11 @@ def test_redirection(self):
             '/es-PE/firefox/addon/1': '/es/firefox/addon/1',
             # /developers doesn't get an app.
             '/developers': '/en-US/developers',
+            # Shorter locales redirects to long form
```
Comment (KevinMind) [inline src/olympia/amo/tests/test_url_prefix.py:41]: Curious how it would handle invalid urls like 

`/-pt/developers`

or 

`/pt--PT/developers`

or

`/-/developers`

Could you add some of those tests to verify sane handling of weird edge cases?

Comment (diox) [reply src/olympia/amo/tests/test_url_prefix.py:41]: addons-server handles all of these poorly (adding `/en-US/firefox/` prefix to all )... but adding a test would be confusing, because none of them match the regexp set in nginx to redirect traffic to addons-server, so they would all be handled by addons-frontend, which _does_ handle them somewhat correctly, redirecting to either `/en-US/developers/` or `/pt-PT/developers/`

Tbh we don't have to care at all - those are not supposed to work anyway.

File: src/olympia/amo/urlresolvers.py (hunk 2 above)
Comment (diox) [reply src/olympia/amo/urlresolvers.py]: c25a40594bb1a88268eb63a7b7ba347c9cdd6e2b

File: src/olympia/amo/tests/test_url_prefix.py (hunk 3 above)
Comment (KevinMind) [reply src/olympia/amo/tests/test_url_prefix.py:41]: Oh man so we have 3 different sources of redirection? Fun.

**Review by KevinMind — APPROVED** [review]
Looks good.
